src/lxc/terminal.c

```diff
@@ -861,13 +861,26 @@ int lxc_terminal_create_log_file(struct lxc_terminal *terminal)
 int lxc_terminal_create(struct lxc_terminal *terminal)
 {
 	int ret;
+	struct termios oldtios;
 
 	ret = openpty(&terminal->master, &terminal->slave, terminal->name, NULL, NULL);
 	if (ret < 0) {
 		SYSERROR("Failed to open terminal");
 		return -1;
 	}
 
+	ret = tcgetattr(terminal->master, &oldtios);
+	if (ret < 0) {
+		SYSERROR("Failed to get flag on terminal master");
+		goto err;
+	}
+	oldtios.c_oflag &= ~ONLCR;
+	ret = tcsetattr(terminal->master, TCSAFLUSH, &oldtios);
+	if (ret < 0) {
+		SYSERROR("Failed to set flag on terminal master");
+		goto err;
+	}
+
 	ret = fcntl(terminal->master, F_SETFD, FD_CLOEXEC);
 	if (ret < 0) {
 		SYSERROR("Failed to set FD_CLOEXEC flag on terminal master");
```

Comment:
I think this should be done in `lxc_setup_tios()`. Then after you do `lxc_terminal_create()` in e.g. `attach.c()` you should call `lxc_setup_tios()`. :)

Reply:
Tks, I have update.